src/primitives/hrp.rs

```diff
@@ -148,6 +148,14 @@ impl Hrp {
     #[inline]
     pub fn to_lowercase(&self) -> String { self.lowercase_char_iter().collect() }
 
+    /// Returns this human-readable part as bytes.
+    #[inline]
+    pub fn as_bytes(&self) -> &[u8] { &self.buf }
+
+    /// Returns this human-readable part as str.
+    #[inline]
+    pub fn as_str(&self) -> Result<&str, str::Utf8Error> { str::from_utf8(&self.buf) }
```

Comment:
We should unwrap this internally and not return a Result. It is impossible for `str::from_utf8` to fail because we check that our HRPs are valid ASCII when we construct them.

Reply:
Ha, we were reviewing at the same time :)

Reply:
You mean `expect` right not unwrap?

```rust
    pub fn as_str(&self) -> &str { str::from_utf8(&self.buf[..self.size]).expect("we only store ASCII bytes") }
```